src/skeleplex/graph/modify_graph.py

```diff
@@ -0,0 +1,360 @@
+                spline_points = np.vstack((points1, points2))
+            #sanity check
+            if np.allclose(spline_points[-1], end_node):
+                logger.info('sanity check passed')    
+
+                
+            _, idx = np.unique(spline_points, axis=0, return_index=True)
+            spline_points = spline_points[np.sort(idx)]
+            spline = B3Spline.from_points(spline_points)
+            merge_attributes[key] = spline
+            merge_attributes[EDGE_COORDINATES_KEY] = spline_points
+        if key == START_NODE_KEY:
+            merge_attributes[key] = n1
+        if key == END_NODE_KEY:
+            merge_attributes[key] = n2
+        if key == GENERATION_KEY:
+            merge_attributes[key] = edge_attributes1[key]
+        
+        if key == LENGTH_KEY:
+            merge_attributes[key] = merge_attributes[EDGE_SPLINE_KEY].arc_length
+
+        if key not in  [VALIDATED_KEY, 
+                            EDGE_COORDINATES_KEY, 
+                            EDGE_SPLINE_KEY, 
+                            START_NODE_KEY, 
+                            END_NODE_KEY, 
+                            GENERATION_KEY,
+                            LENGTH_KEY]:
+            logger.warning(('Warning: Attribute {} not merged. '.format(key) ,
+                            'Consider recomputing.'))
+            
+    graph.add_edge(*merge_edge, **merge_attributes)
+    return graph
+
+def delete_edge(SkeletonGraph_obj:SkeletonGraph, edge: Tuple[int, int]):
+        """delete edge."""
+
+        #check if directed
+        if not SkeletonGraph_obj.graph.is_directed():
+            ValueError('Graph is not directed. Convert to directed graph.')
+        #copy graph
+        graph = SkeletonGraph_obj.graph.copy()
+        graph.remove_edge(*edge)
+
+        #detect all changes
+        changed_edges = set(SkeletonGraph_obj.graph.edges) - set(graph.edges)
+        for edge in changed_edges:
+            for node in edge:
+                if graph.degree(node) == 0:
+                    graph.remove_node(node)
+                #merge edges if node has degree 2
+                elif graph.degree(node) == 2:
+                    #merge
+                    in_edge = list(graph.in_edges(node))
+                    out_edge = list(graph.out_edges(node))
+                    if len(in_edge) == 0:
+                        raise ValueError(('Deleting the edge would break the graph'),
+                                         'Are you trying to delete the origin?')
+                        
+                    graph = merge_edge(graph,in_edge[0][0], node, out_edge[0][1])
+                    logger.info('merge')
+
+
+        #check if graph is still connected, if not remove orphaned nodes
+        SkeletonGraph_obj.graph.remove_nodes_from(list(nx.isolates(SkeletonGraph_obj.graph)))
+        SkeletonGraph_obj.graph = graph
+
+
+def length_pruning(SkeletonGraph_obj:SkeletonGraph, length_threshold:int):
```

Comment:
I prefer to use lower case variables names.

```suggestion
def length_pruning(skeleton_graph: SkeletonGraph, length_threshold:int):
```